app/models/template_input.rb

```diff
@@ -26,6 +26,7 @@ class UnsatisfiedRequiredInput < ::Foreman::Exception
   validates :variable_name, :presence => { :if => :variable_template_input? }
   validates :puppet_parameter_name, :puppet_class_name, :presence => { :if => :puppet_parameter_template_input? }
   validates :value_type, inclusion: { in: VALUE_TYPE }
+  validates :default, inclusion: { in: :possible_values }
```

Comment:
I believe we should validate only if we have options defined. For free text input this validation is failing - see test failures.

Reply:
Thanks! Now tests are green.